openwebstart/src/main/java/com/openwebstart/app/icon/ApplicationIconCache.java

```diff
@@ -50,7 +44,7 @@ public ApplicationIconCache(final Executor executor) {
 
     public synchronized void triggerDownload(final Application application, final IconDimensions dimension, final boolean reloadIfAlreadyInCache) {
         if (reloadIfAlreadyInCache || !isInCache(application, dimension)) {
-            executor.execute(() -> loadAndAdd(application, dimension));
+            getDaemonExecutorService().execute(() -> loadAndAdd(application, dimension));
```

Comment:
It won't be a problem to have this one as a non daemon executor. I prefer to change as less behaviour as possible with this PR so we should change it here to non daemon.